### PR TITLE
Rules: New options replace_name to replace ${name} in attributes

### PR DESCRIPTION
**Pull request by AlexanderS**
If you use the regex feature of Rules/Defaults you may need the real name
of the matched entry in an attribute (for example home of POSIXUser).
You can now enable replace_name for rules or defaults and ${name} will be
replaces in the attribues of the Element before adding them to the target
entry.

This allows you to write something like that in Defaults to assing a default
home directory somewhere else to all users with unset home:

    <POSIXUser name='.*' home='/somewhere/${name}'/>

### CODE REVIEW

File: doc/server/plugins/generators/rules.txt [hunk 1]
```diff
@@ -512,3 +512,23 @@ you'd have to explicitly specify ``<Service name="bcfg2.*".../>``.
 
 Note that only one Rule can apply to any abstract entry, so you cannot
 specify multiple regexes to match the same rule.
+
+Replacing the name of the Entry in Attribues
```
Comment (solj) [inline doc/server/plugins/generators/rules.txt]: Should be Attributes

File: doc/server/plugins/generators/rules.txt [hunk 2]
```diff
@@ -512,3 +512,23 @@ you'd have to explicitly specify ``<Service name="bcfg2.*".../>``.
...
+============================================
+
+If you are using regular expressions to match the abstract configuration
+entries, you may need the concrete name of the entry in some attributes.
+To use this feature, you have to enable it. It is only usefull, if used
```
Comment (solj) [inline doc/server/plugins/generators/rules.txt]: another typo "useful"

File: doc/server/plugins/generators/rules.txt [hunk 3]
```diff
@@ -512,3 +512,23 @@ you'd have to explicitly specify ``<Service name="bcfg2.*".../>``.
...
+together with regex matching. ::
+
+    [rules]
+    regex = yes
+    replace_name = yes
+
+You now can write somehing like that in your xml file:
```
Comment (solj) [inline doc/server/plugins/generators/rules.txt]: typo "something"

**Comment by solj** [comment]
This looks good to me at first glance. Can you also mention the feature in the release notes?

**Comment by AlexanderS** [comment]
Fixed the typos and added a line to the release notes.

**Comment by solj** [comment]
:+1:

**Comment by stpierre** [comment]
This seems like it won't work for (or at least will conflict with) Rules or Defaults files that are Genshi templates, since ``${name}`` will look like a Genshi variable. So depending on the order of evaluation, either a) I can no longer use ``name`` as a variable in my Genshi template; or b) I can't use replace_name in Genshi templates. Either way it's unexpected.

I'd suggest using a different format for the name token. ``%(name)s`` would be easy enough from a code point of view, but I'm open to other suggestions as long as they don't conflict with Genshi templating.

Additionally, I can see the use case for this in Defaults, but not so much in Rules. (I've never liked the Rules plugin, though, so I might be biased.) What's the use case in Rules?

**Comment by AlexanderS** [comment]
I simply replaced the delimiter for `string.Template` with `%`, so that you can use `%name` or `%{name}`. This has the advantage over the classic `%` format strings, that it is more robust and does not fail if extra placeholders are in the string.

**Pull request merged**